wallet/wallet-cli/src/config.rs

```diff
@@ -0,0 +1,150 @@
+    #[clap(long)]
+    pub rpc_username: Option<String>,
+
+    /// RPC password (either provide a username and password, or use a cookie file. You cannot use both)
+    #[clap(long)]
+    pub rpc_password: Option<String>,
+
+    /// vi input mode
+    #[clap(long)]
+    pub vi_mode: bool,
+}
+
+#[derive(Debug)]
+pub struct WalletCliConfig {
+    pub chain_type: ChainType,
+    pub data_dir: PathBuf,
+    pub wallet_file: PathBuf,
+    pub rpc_address: SocketAddr,
+    pub rpc_username: String,
+    pub rpc_password: String,
+    pub vi_mode: bool,
+}
+
+const DEFAULT_WALLETS_DIR: &str = "wallets";
+const DEFAULT_WALLET_NAME: &str = "wallet";
+
+const COOKIE_FILENAME: &str = ".cookie";
+
+fn load_cookie(path: impl AsRef<Path>) -> Result<(String, String), WalletCliError> {
+    let content = std::fs::read_to_string(path.as_ref())
+        .map_err(|e| WalletCliError::CookieFileReadError(path.as_ref().to_owned(), e))?;
+    let (username, password) = content.split_once(':').ok_or(WalletCliError::InvalidConfig(
+        format!("Invalid cookie file {:?}: ':' not found", path.as_ref()),
+    ))?;
+    Ok((username.to_owned(), password.to_owned()))
+}
+
+impl WalletCliConfig {
+    pub fn from_args(args: WalletCliArgs) -> Result<WalletCliConfig, WalletCliError> {
+        let WalletCliArgs {
+            network,
+            wallets_dir,
+            wallet_name,
+            rpc_address,
+            rpc_cookie_file,
+            rpc_username,
+            rpc_password,
+            vi_mode,
+        } = args;
+
+        let chain_type = match network {
+            Network::Mainnet => ChainType::Mainnet,
+            Network::Testnet => ChainType::Testnet,
+            Network::Regtest => ChainType::Regtest,
+            Network::Signet => ChainType::Signet,
+        };
+
+        let data_dir = prepare_data_dir(
+            || default_data_dir_for_chain(chain_type.name()).join(DEFAULT_WALLETS_DIR),
+            &wallets_dir,
+        )
+        .map_err(WalletCliError::PrepareData)?;
+        let wallet_file = data_dir.join(wallet_name.unwrap_or(DEFAULT_WALLET_NAME.into()));
+
+        // TODO: Use the constant with the node
+        let default_http_rpc_addr = || SocketAddr::from_str("127.0.0.1:3030").expect("Can't fail");
+        let rpc_address = rpc_address.unwrap_or_else(default_http_rpc_addr);
+
+        let (rpc_username, rpc_password) = match (rpc_cookie_file, rpc_username, rpc_password) {
+            (None, None, None) => {
+                load_cookie(default_data_dir_for_chain(chain_type.name()).join(COOKIE_FILENAME))?
+            }
+            (Some(cookie_path), None, None) => load_cookie(cookie_path)?,
+            (None, Some(username), Some(password)) => (username, password),
+            _ => {
+                return Err(WalletCliError::InvalidConfig(
+                    "Invalid RPC cookie/username/password combination".to_owned(),
+                ))
+            }
+        };
```

Comment:
Let's forget the part of the wallet file for now, but the wallet file is also part of this discussion.

Something to keep in mind, is that the wallet initialization from a utilization point of view shouldn't be in the wallet-cli crate, but should be in a separate crate, which can be used by any wallet implementation we create in the future. Imagine in the future we create a GUI wallet. Everything should be common between those two, and the only difference is the configuration we input into those two. 

So what should be in the wallet CLI is only everything related to CLI specifically, not to the wallet. Like prompts, messages, commands, etc. Things like initialization, creation, etc, should all go to a separate crate that is used in this crate. In the future, once we establish the GUI wallet, we just import that crate and we use it, and spend time only on designing the GUI.

Reply:
Okay, thanks. I will keep that in mind.